pkg/controller/plan/controller.go

```diff
@@ -189,7 +189,7 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 	}()
 
 	// Don't reconcile if the plan is archived.
-	if plan.Spec.Archived && plan.Status.HasCondition(Archived) {
+	if r.shouldAbortArchived(plan) {
```

Comment:
Please don't obscure these simple conditions behind method calls. It makes it harder to read when you have to jump to a method to find out what the condition is.